File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.java
class Solution {
    public List<Integer> intersection(int[][] nums) {
        Map<Integer, Integer> hashMap = new HashMap<>();
        for (int[] num : nums) {
            for (int element : num) {
                Integer value = hashMap.getOrDefault(element, 0);
                hashMap.put(element, value+1);
            }
        }
        List<Integer> answer = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : hashMap.entrySet()) {
            if (entry.getValue().equals(nums.length)) {
                answer.add(entry.getKey());
            }
        }
        Collections.sort(answer);
        return answer;
    }
}

File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.go
func intersection(nums [][]int) []int {
	hashMap := make(map[int]int)
	for _, num := range nums {
		for _, element := range num {
			hashMap[element]++
		}
	}
	answer := make([]int, 0)
	for k, v := range hashMap {
		if v == len(nums) {
			answer = append(answer, k)
		}
	}
	sort.Ints(answer)
	return answer
}
